File: 241126/재귀함수를 이용한 3n + 1 수열/3n-plus-1-sequence-with-recursive-function.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.println(sums(n));
    }
    public static int sums(int number){
        if(number == 1){
            return 0;
        }
        if(number % 2 == 1){
            return sums(number*3 + 1) + 1;
        }else{
            return sums(number/2) + 1;
        }
    }
}

File: 241126/재귀함수를 이용한 3n + 1 수열/3n-plus-1-sequence-with-recursive-function.py
n = int(input())
def sums(n):
    if n == 1:
        return 0
    
    if n % 2 == 0:
        return sums(n//2) + 1
    else:
        return sums(n*3+1) + 1
print(sums(n))